packages/wrangler/src/__tests__/pages/project-upload.test.ts

```diff
@@ -516,6 +516,67 @@ describe("pages project upload", () => {
 	`);
 	});
 
+	it("should handle a very large number of assets", async () => {
+		const assets = new Set<string>();
+		// Create a large number of asset files to upload
+		for (let i = 0; i < 10_000; i++) {
+			const path = `file-${i}.txt`;
+			const content = `contents of file-${i}.txt`;
+			assets.add(content);
+			writeFileSync(path, content);
```

Comment:
any memory concerns here regarding writing 10k files? I understand the file size is small, but worth asking nonetheless

Reply:
I'm not worried. The total memory used in this is going to be less than a megabyte.